src/dfx/src/lib/extension/manifest/extension.rs

```diff
@@ -14,15 +21,36 @@ pub struct ExtensionManifest {
     pub categories: Vec<String>,
     pub keywords: Option<Vec<String>>,
     pub description: Option<String>,
-    pub commands: JsonValue,
+    pub subcommands: JsonValue, // TODO: awaiting https://dfinity.atlassian.net/browse/SDK-599
     pub dependencies: Option<HashMap<String, String>>,
 }
 
 impl Display for ExtensionManifest {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let Ok(json) = serde_json::to_string_pretty(self) else {
+        let Ok(s) = toml::to_string_pretty(self) else {
             return Err(std::fmt::Error)
         };
-        write!(f, "{}", json)
+        write!(f, "{}", s)
+    }
+}
+
+impl ExtensionManifest {
+    pub fn from_extension_directory(path: PathBuf) -> Result<Self, ExtensionError> {
+        let manifest_path = path.join(MANIFEST_FILE_NAME);
```

Comment:
Given that the only caller is:
```
                let text = match ExtensionManifest::from_extension_directory(
                    self.get_extension_directory(&extension.name),
                ) {
```

Recommend something like this:

```suggestion
    pub fn new(name: String, extensions_root_dir: &Path) -> Result<Self, ExtensionError> {
        let manifest_path = extensions_root_dir.join(name).join(MANIFEST_FILE_NAME);
```
and call with 
```
ExtensionManifest::new(&extension.name, self.dir)
```